ginga/misc/plugins/Cuts.py

```diff
@@ -319,23 +330,54 @@ def stop(self):
 
     def _plotpoints(self, obj, color):
         image = self.fitsimage.get_image()
-        # Get points on the line
-        if obj.kind == 'line':
-            points = image.get_pixels_on_line(int(obj.x1), int(obj.y1),
-                                              int(obj.x2), int(obj.y2))
-        elif obj.kind in ('path', 'freepath'):
-            points = []
-            x1, y1 = obj.points[0]
-            for x2, y2 in obj.points[1:]:
-                pts = image.get_pixels_on_line(int(x1), int(y1),
-                                               int(x2), int(y2))
-                # don't repeat last point when adding next segment
-                points.extend(pts[:-1])
-                x1, y1 = x2, y2
-
-        points = numpy.array(points)
-        self.plot.cuts(points, xtitle="Line Index", ytitle="Pixel Value",
-                       color=color)
+
+        if self.cut_axis_enabled:
+            path = image.get('path', None)
+            hdus = pyfits.open(path, 'readonly')
+            phd = hdus[0].data
+
+            if self.cut_axis_idx == 0:      # NAXIS3
+                # Get points on the line
+                if obj.kind == 'line':
+                    coords = image.get_pixels_on_line(int(obj.x1), int(obj.y1),
+                                                      int(obj.x2), int(obj.y2), getvalues=False)
+                elif obj.kind in ('path', 'freepath'):
+                    coords = []
+                    x1, y1 = obj.points[0]
+                    for x2, y2 in obj.points[1:]:
+                        coords = image.get_pixels_on_line(int(x1), int(y1),
+                                                          int(x2), int(y2), getvalues=False)
+                        # don't repeat last point when adding next segment
+                        coords.extend(coords[:-1])
+                        x1, y1 = x2, y2
+
+                coords = np.array(coords)
+                slit = phd[:, coords[:, 1], coords[:, 0]]
+
+                plt.imshow(slit, cmap=plt.get_cmap('gray'), interpolation='nearest', origin='lower')
```

Comment:
Unfortunately, self.plot.ax.imshow(...) does not update the plot. I was wondering if the plot area could be split into two but that would scale down the plot.